posthog/api/team.py

```diff
@@ -506,6 +514,18 @@ def get_serializer_class(self) -> type[serializers.BaseSerializer]:
             return TeamBasicSerializer
         return super().get_serializer_class()
 
+    def dangerously_get_required_scopes(self, request, view) -> list[str]:
+        # If the request only contains config fields, require read:team scope
+        # Otherwise, require write:team scope (handled by APIScopePermission)
+        if self.action == "partial_update":
+            request_fields = set(request.data.keys())
+            non_config_fields = request_fields - set(CONFIG_FIELDS)
+            if not non_config_fields:
```

Comment:
Let's defined `CONFIG_FIELDS_SET = set(CONFIG_FIELDS)` globally, so that we don't recompute every single time. Then this _could_ be a little more readable (subjectively):
```suggestion
            if not (CONFIG_FIELDS_SET & set(request.data.keys()):
```

Reply:
Yeah can do 👍